### DISCUSSION TITLE
Nuxt3 Options API: Why I can't use composables inside setup function?

### DISCUSSION THREAD

**Discussion by Blabter** [nuxt/nuxt#26255] [Questions]
I have Options API component:

```
export default defineComponent({
  name: "mycomponent",
  async setup() {
    console.log(await useFetch('http://localhost/api/data'));
  },
})
```

When I'm trying to run this code Nuxt3 give me error:

```
500
[nuxt] A composable that requires access to the Nuxt instance was called outside of a plugin, Nuxt hook, Nuxt middleware, or Vue setup function. This is probably not a Nuxt bug. Find out more at `https://nuxt.com/docs/guide/concepts/auto-imports#vue-and-nuxt-composables`.
```

Error says I must call this composable from Vue setup function, but I'm already calling this composable from setup function. What's the problem?

**Comment by huang-julien** [comment]
can you try taking out the await useFetch from the console.log ? Vue and Nuxt does file transformation to keep the context when awaiting promises.
